app/models/user.rb

```diff
@@ -1,5 +1,13 @@
 class User < ApplicationRecord
-  has_many :posts, dependent: :destroy
-  has_many :likes, dependent: :destroy
-  has_many :comments, dependent: :destroy
+  has_many :comments
+  has_many :posts
+  has_many :likes
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_count
+    update(posts_count: posts.count)
+  end
```

Comment:
- [ ] This method belongs in the Post model, not the User model. I can see you are calling this method from the Post model, but I think it would more organized to move the method to the Post model since the callback that triggers it is in the Post model, not the User model.
- [ ] Remember to also add the `after_create` callback so that this method runs automatically whenever a post is saved.
- [ ] When you move the method to the Post model you also need to refactor it to update the posts_counter of the user. You can use this.

  ```
  def update_posts_counter
    author.increment!(:posts_counter)
  end
  ```